homeassistant/components/lock/isy994.py

```diff
@@ -0,0 +1,91 @@
+"""
+Support for ISY994 locks.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/isy994/
+"""
+import logging
+
+from homeassistant.components.isy994 import (
+    HIDDEN_STRING, ISY, SENSOR_STRING, ISYDeviceABC)
+from homeassistant.const import STATE_LOCKED, STATE_UNLOCKED
+
+
+# The frontend doesn't seem to fully support the open and closed states yet.
+# Once it does, the HA.doors programs should report open and closed instead of
+# off and on. It appears that on should be unlocked and off should be locked.
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ISY994 platform."""
+    # pylint: disable=too-many-locals
+    logger = logging.getLogger(__name__)
+    devs = []
+    # verify connection
+    if ISY is None or not ISY.connected:
+        logger.error('A connection has not been made to the ISY controller.')
+        return False
+
+    # Import not dimmable nodes and groups
+    for (path, node) in ISY.nodes:
+        if not node.dimmable and SENSOR_STRING not in node.name:
```

Comment:
So I have an ISY - I have some custom code for configuration.

I think we should just do `DOMAIN in node.name` this will allow you to put switches in switches, sensors in sensors, etc.